libraries/chain/database.cpp

```diff
@@ -3402,21 +3402,30 @@ void database::update_global_dynamic_data( const signed_block& b )
       for( uint32_t i = 0; i < missed_blocks; ++i )
       {
          const auto& witness_missed = get_witness( get_scheduled_witness( i + 1 ) );
-         if(  witness_missed.owner != b.witness )
+
+         modify( witness_missed, [&]( witness_object& w )
          {
-            modify( witness_missed, [&]( witness_object& w )
+            w.current_run = 0;
+            if( witness_missed.owner != b.witness )
             {
+               //
+               // total_missed does not increment when witness_missed.owner == b.witness
+               //    because a low total_missed is a "prestige" item and a witness that
+               //    restarts a dead network is "rewarded" by not having total_missed
+               //    increase for any blocks they missed in the gap.
+               // Also, this prevents initminer from having a large total_missed.
+               //
+
                w.total_missed++;
```

Comment:
This should be wrapped in the `if( witness_missed.owner != b.witness )` conditional to preserve existing behavior.